bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/data/ObservationProvider.scala

```diff
@@ -181,7 +184,15 @@ case class PositionProvider(ctx: QvContext, base: ObservationProvider) extends O
     _observations = base.observations.map(o => {
       if (NonSiderealCache.isHorizonsTarget(o)) {
         val pos = NonSiderealCache.get(ctx.site, ctx.referenceDate, o)
-        val newTarget = o.getTargetEnvironment.getBase.clone()
+
+        // The check above ensures that this is a nonsidereal observation, which means it must have
+        // a single-target asterism. If this is not the case it means the model has changed.
+        val newTarget = o
+          .getTargetEnvironment
+          .getAsterism
+          .getNonSiderealSpTarget
+          .getOrElse(sys.error("The asterism is not a single nonsidereal target."))
+
```

Comment:
I don't think there is a much better way do this without bubbling up the `Option` which as you said will bring a bunch of other changes